web/src/engine/websites/MangaGecko.ts

```diff
@@ -0,0 +1,47 @@
+import { Tags } from '../Tags';
+import icon from './MangaGecko.webp';
+import { Chapter, DecoratableMangaScraper, type MangaPlugin, type Manga } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchCSS, FetchRequest } from '../FetchProvider';
+
+@Common.MangaCSS(/^{origin}\/manga\/[^/]+\/$/, 'div.main-head h1[itemprop="name"]')
+@Common.PagesSinglePageCSS('section.page-in div img[onerror]')
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('mangagecko', `MangaGecko`, 'https://www.mgeko.com', Tags.Media.Manga, Tags.Media.Manhua, Tags.Media.Manhwa, Tags.Language.English, Tags.Source.Aggregator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+
+        //get pagination because it never ends otherwise
+        const request = new FetchRequest(`${this.URI.origin}/browse-comics/?results=1`);
+        const data = await FetchCSS<HTMLAnchorElement>(request, 'ul.pagination li:not([class]) a');
+        const pageMax = parseInt(data[0].text.trim());
+        const mangaList: Manga[] = [];
+
+        let reducer = Promise.resolve();
+        for (let page = 1; page <= pageMax; page++) {
+            await reducer;
+            reducer = new Promise(resolve => setTimeout(resolve, 250));
```

Comment:
No need for intermediate `reducer` variable, just await the new `Promise`, the initial 250ms delay should hardly be noticeable for the user.
Alternatively, use `Taskpool` with appropriate rate limiter.